### PR TITLE
Improve docs and hooks-example for `useMap` and `MapProvider`

### PR DESCRIPTION
**Pull request by tordans**
I took some time to learn `useMap` and `MapProvider` today. The changes in this PR represent what I would have needed to understand the behavior of `useMap` and `MapProvider` faster. I ran into issues because I picked a name for my map id that collided with in import. And I used `useMap` right in the component which rendered the MapProvider, which – seems obvious now – does not work.

The commits are kept tiny to give them all a "why I added this" commit message.

### CODE REVIEW

**Comment by benderlidze** [comment]
And `const {current: map} = useMap();` map is always undefined. It only works if you use it with the map id only. 
Like `const { current: map, mymap } = useMap();`  mymap is ok, map - undefined. Here is a link to the discussion at Mapbox https://github.com/mapbox/mapbox-gl-js/issues/11816

**Comment by tordans** [comment]
> 

@benderlidze it does work when used in Control2 which is a child of Map. I expected it to, since that is the only place where current is defined at all.

Console:

image


Diff to this PR for local Testing:

```diff
diff --git a/examples/get-started/hook/controls.js b/examples/get-started/hook/controls.js
index e473accd..42503d6d 100644
--- a/examples/get-started/hook/controls.js
+++ b/examples/get-started/hook/controls.js
@@ -4,7 +4,7 @@ import {useCallback, useState, useEffect} from 'react';
 import {useMap} from 'react-map-gl';

 export default function Controls() {
-  const demo = useMap();
+  const {current: demo} = useMap();
   console.log('Controls', {demo});
   // First render:
   // {
diff --git a/examples/get-started/hook/controls2.js b/examples/
get-started/hook/controls2.js
index 619c5847..79e80f8d 100644
--- a/examples/get-started/hook/controls2.js
+++ b/examples/get-started/hook/controls2.js
@@ -9,7 +9,7 @@ import {useMap} from 'react-map-gl';
 // that references the containing map.
 // See /docs/api-reference/use-map.md
 export default function Controls2() {
-  const demo = useMap();
+  const {current: demo} = useMap();
   console.log('Controls2', {demo});
   // First render:
   // {
diff --git a/examples/get-started/hook/map.js b/examples/get-st
arted/hook/map.js
index 0712d16e..95496d7f 100644
--- a/examples/get-started/hook/map.js
+++ b/examples/get-started/hook/map.js
@@ -7,7 +7,7 @@ import Controls2 from './controls2';
 const MAPBOX_TOKEN = ''; // Set your mapbox token here

 export default function MapView() {
-  const demo = useMap();
+  const {current: demo} = useMap();
   console.log('MapView', {demo});
   // First render:
   // {
```

**Review by Pessimistress — COMMENTED** [review]
I have no objection to the doc updates, but I don't think the `console.log` statements belong in the examples. If you think they are helpful, one way to do it is to add the code snippet to the documentation page under `useMap`, then link to the documentation from the example apps.

**Comment by tordans** [comment]
I converted this do a draft since it will need to be re applied to the latest version of the docs and code.

**Comment by Bonkles** [comment]
@tordans this PR saved my bacon in a side project I just started. These doc additions about the map id being crucial escaped me too!

File: examples/get-started/hook/controls.jsx [hunk 1]
```diff
@@ -4,14 +4,38 @@ import {useCallback, useState, useEffect} from 'react';
...
+   * ```
+   * ### Second render:
+   * ```
+   * {
+   *   "demo": {
+   *     "current": undefined,
+   *     "mymap": {...} // See https://visgl.github.io/react-map-gl/docs/api-reference/types#mapref
+   *   }
+   * }
+   * ```
```
Comment (Pessimistress) [inline examples/get-started/hook/controls.jsx:29]:
```suggestion
   * ## This is how `useMap` works:
   * ```
   * const maps = useMap();
   * console.log('Controls useMap()', maps);
   * ```
   * ### First render:
   * ```
   * {
   *     "current": undefined
   * }
   * ```
   * ### Second render:
   * ```
   * {
   *     "current": undefined,
   *     "mymap": {...} // See https://visgl.github.io/react-map-gl/docs/api-reference/types#mapref
   * }
   * ```
```

File: examples/get-started/hook/controls2.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+   * ```
+   * ### First render:
+   * ```
+   * {
+   *     "demo": {
+   *       "current": {...},
+   *       "mymap": {...}
+   *     }
+   * }
+   * ```
```
Comment (Pessimistress) [inline examples/get-started/hook/controls2.js:22]:
```suggestion
   * const maps = useMap();
   * console.log('Controls2 useMap()', maps);
   * ```
   * ### First render:
   * ```
   * {
   *     "current": {...}, // this is the same as `mymap`
   *     "mymap": {...}
   * }
   * ```
```

File: examples/get-started/hook/map.jsx [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
+   *     }
+   * }
+   * Second render:
+   * {
+   *     "demo": {
+   *         "current": undefined,
+   *         "mymap": {...} // See https://visgl.github.io/react-map-gl/docs/api-reference/types#mapref
+   *     }
+   * }
+   * ```
```
Comment (Pessimistress) [inline examples/get-started/hook/map.jsx:30]:
```suggestion
   * ```
   * const maps = useMap();
   * console.log('MapView useMap()', maps);
   * ```
   * ### First render:
   * ```
   * {
   *     "current": undefined
   * }
   * Second render:
   * {
   *     "current": undefined,
   *     "mymap": {...} // See https://visgl.github.io/react-map-gl/docs/api-reference/types#mapref
   * }
   * ```
```

**Comment by tordans** [comment]
@Pessimistress thanks for the review, those changes make it easier to understand! I rebased to resolve the conflicts. Anything else I can do?

**Review by Pessimistress — APPROVED** [review]
Thank you!

**Comment by Pessimistress** [comment]
@tordans Could you fix the lint errors introduced by your change?

**Comment by tordans** [comment]
@Pessimistress I fixed the linter issues and rebased on current master. But the action requires your approval before running.

**Pull request merged**